flixel/graphics/frames/bmfontutils/BMFont.hx

```diff
@@ -0,0 +1,110 @@
+package flixel.graphics.frames.bmfontutils;
+
+typedef BMFontInfoBlock =
+{
+	public var fontSize:Null<Int>;
+	public var smooth:Bool;
+	public var unicode:Bool;
+	public var italic:Bool;
+	public var bold:Bool;
+	public var fixedHeight:Bool;
+	public var charSet:String;
+	public var stretchH:Null<Int>;
+	public var aa:Null<Int>;
+	public var paddingUp:Null<Int>;
+	public var paddingRight:Null<Int>;
+	public var paddingDown:Null<Int>;
+	public var paddingLeft:Null<Int>;
+	public var spacingHoriz:Null<Int>;
+	public var spacingVert:Null<Int>;
+	public var outline:Null<Int>;
+	public var fontName:String;
```

Comment:
Are these all really nullable in the data? or is this so you can create an instance in the constructor? Couple notes:
1. I think you mean to use `@:optional var fontSize:Null<Int>;` which is similar to `Null<Int>` except the field can be ignored when creating instances
2. Can all of these fields be ommitted from the data? or is it because you're constructing this from an empty struct.

I these typedefs should match the underlying format. if an int should be expected to be in the data, the typedef should mark it as an Int, not a Null<Int> or @:optional Int, and vice versa, if an int is optional in the data, it should be an @:optional int

Reply:
additionally, you seem to be assuming most of these fields are not null when you access them, reinforcing my assumption that they are not actually nullable, and you just want to give them default values of null

Reply:
> Are these all really nullable in the data?

They aren't, I completely forgot that `@:optional` existed lol. I've made changes to remove the `Null<T>` types and give them some default values instead, and from my tests on windows and html5 it still seems to work.